Backend/Remora.Discord.Voice/DiscordVoiceClient.cs

```diff
@@ -0,0 +1,1258 @@
+                    channelID
+                );
+
+                _disconnectRequestedSource.Dispose();
+                _disconnectRequestedSource = new CancellationTokenSource();
+
+                var connectionResult = await ConnectAsync(_connectionRequestParameters, ct).ConfigureAwait(false);
+                if (!connectionResult.IsSuccess)
+                {
+                    SendDisconnectVoiceStateUpdate(guildID);
+                    await CleanupAsync().ConfigureAwait(false);
+                    return connectionResult;
+                }
+
+                _runLoopTask = RunnerAsync(_connectionRequestParameters, _disconnectRequestedSource.Token);
+
+                return Result.FromSuccess();
+            }
+            catch (Exception ex)
+            {
+                await CleanupAsync().ConfigureAwait(false);
+                return ex;
+            }
+        }
+
+        /// <summary>
+        /// Stops the voice client.
+        /// </summary>
+        /// <returns>A <see cref="Result"/> representing the outcome of the operation.</returns>
+        public async Task<Result> StopAsync()
+        {
+            try
+            {
+                if (!this.IsRunning)
+                {
+                    return new InvalidOperationError("Already stopped.");
+                }
+
+                await CleanupAsync().ConfigureAwait(false);
+
+                var runLoopTaskResult = await _runLoopTask.ConfigureAwait(false);
+                if (!runLoopTaskResult.IsSuccess)
+                {
+                    return runLoopTaskResult;
+                }
+
+                return Result.FromSuccess();
+            }
+            catch (Exception ex)
+            {
+                return ex;
+            }
+            finally
+            {
+                this.ConnectionStatus = GatewayConnectionStatus.Offline;
+            }
+        }
+
+        /// <summary>
+        /// Enqueues a voice gateway command for sending.
+        /// </summary>
+        /// <typeparam name="TCommand">The type of the command to send.</typeparam>
+        /// <param name="command">The command object.</param>
+        public void EnqueueCommand<TCommand>(TCommand command) where TCommand : IVoiceGatewayCommand
+        {
+            var payload = new VoicePayload<TCommand>(command);
+            _payloadsToSend.Enqueue(payload);
+        }
+
+        /// <summary>
+        /// Transmits audio.
+        /// </summary>
+        /// <param name="audioStream">The audio data to send.</param>
+        /// <param name="speakingFlags">The speaking flags to use.</param>
+        /// <param name="transcoder">The audio transcoder to use.</param>
+        /// <param name="ct">A <see cref="CancellationToken"/> that can be used to stop the operation.</param>
+        /// <returns>A result representing the outcome of the operation.</returns>
+        public async Task<Result> TransmitAudioAsync
+        (
+            Stream audioStream,
```

Comment:
So, if I'm understanding correctly, essentially how this is to be used is you ➜ stream, and so long as you copy data to the stream that's passed, it will continue to play?

Reply:
Yeah, as you've discovered. Data is just read from the stream in chunks of the size that the transcoder requests. And you can call Transmit multiple times in a row with different streams.